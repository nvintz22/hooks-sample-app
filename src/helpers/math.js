export const sum = (x,y) => x + y;
export const subtract = (x,y) => x - y;
export const multiple = (x,y) => x * y;
export const divide = (x,y) => x / y;
