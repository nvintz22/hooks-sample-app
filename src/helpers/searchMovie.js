import Movies from '../db.json'
import { filter } from 'lodash';

export const FindMovies = keyword => {
    if (keyword) {
        const foundMovies = filter(Movies, qry => qry.title.includes(keyword))
        return foundMovies
    }
    return Movies
}