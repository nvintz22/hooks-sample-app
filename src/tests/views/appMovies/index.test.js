import React from 'react'
import enzymeSerializer from 'enzyme-to-json/serializer'
import { mount } from 'enzyme'
import { FindMovies } from '../../../helpers/searchMovie'
import Movies from '../../../views/appMovies';

expect.addSnapshotSerializer(enzymeSerializer)
expect.addSnapshotSerializer({
    test: val => val.title && val.cover,
    print: val => `${val.cover} ${val.title}`
})

describe('Movies' , () => {
    it('should return all movies', () => {
        const Movies = FindMovies()
        expect(Movies).toMatchSnapshot()
    })
 
    it('should return avengers movie', () => {
        const Avengers = FindMovies('Avengers')
        expect(Avengers).toMatchSnapshot()
    })

    it('should return movie list', () => {
        const MoviesComponent = mount(<Movies />)
        expect(MoviesComponent).toMatchSnapshot()
    })
})