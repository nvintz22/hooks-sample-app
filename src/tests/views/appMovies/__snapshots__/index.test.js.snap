// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies should return all movies 1`] = `
Array [
  alladin.jpg Alladin,
  avengers.jpg Avengers End Game,
  captain-marvel.jpg Captain Marvel,
  dark-phoenix.jpg Dark Phoenix,
  glass.jpg Glass,
  shazam.jpg Shazam,
  spiderman-far-from-home.jpg Spiderman Far from home,
  star-wars.jpg Star Wars,
  toy-story-4.jpg Toy Story 4,
]
`;

exports[`Movies should return avengers movie 1`] = `
Array [
  avengers.jpg Avengers End Game,
]
`;

exports[`Movies should return movie list 1`] = `
<AppMovies>
  <Styled(div)
    className="group"
  >
    <div
      className="group css-1c687hf"
    >
      <Styled(input)
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      >
        <input
          className="css-zs2jp2"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </Styled(input)>
      <span
        className="highlight"
      />
      <Styled(div)
        className="bar"
      >
        <div
          className="bar css-utljbf"
        />
      </Styled(div)>
      <Styled(label)>
        <label
          className="css-1odazk5"
        >
          Search
        </label>
      </Styled(label)>
    </div>
  </Styled(div)>
  <ul />
</AppMovies>
`;
